content_sync/tasks.py

```diff
@@ -392,3 +398,120 @@ def check_incomplete_publish_build_statuses():
+            for website in sites:
+                version = (
+                    VERSION_DRAFT
+                    if (
+                        website.draft_publish_date is not None
+                        and website.publish_date is None
+                    )
+                    else VERSION_LIVE
+                )
+                bulk_update.queue(
+                    WebsiteContent(
+                        website=root_website,
+                        type="website",
+                        title=website.title,
+                        dirpath=WEBSITE_LISTING_DIRPATH,
+                        filename=website.short_id,
+                        is_page_content=True,
+                        metadata=get_website_in_root_website_metadata(website, version),
+                    )
+                )
+        website_content = WebsiteContent.objects.filter(
+            website=root_website, type="website"
+        )
+        with ContentSyncState.objects.bulk_update_or_create_context(
+            ["content", "current_checksum"], match_field="content", batch_size=100
+        ) as bulk_update:
+            for content in website_content:
+                bulk_update.queue(
+                    ContentSyncState(
+                        content=content, current_checksum=content.calculate_checksum()
+                    )
+                )
+        backend = api.get_sync_backend(website=root_website)
+        backend.sync_all_content_to_backend(query_set=website_content)
+
+
+@app.task(acks_late=True)
+def update_website_in_root_website(website, version):
+    """Create or update a WebsiteContent object of type website in the website denoted by settings.ROOT_WEBSITE_NAME"""
+    if (
+        website.name != settings.ROOT_WEBSITE_NAME
+        and WebsiteContent.objects.filter(website=website, type="sitemetadata").exists()
+    ):
+        root_website = Website.objects.get(name=settings.ROOT_WEBSITE_NAME)
+        root_has_unpublished = (
+            root_website.has_unpublished_live
+            if version == VERSION_LIVE
+            else root_website.has_unpublished_draft
+        )
+        (
+            website_content,
+            created,  # pylint:disable=unused-variable
+        ) = WebsiteContent.all_objects.update_or_create(
+            website=root_website,
+            dirpath=WEBSITE_LISTING_DIRPATH,
+            filename=website.short_id,
+            defaults={
+                "title": website.title,
+                "type": "website",
+                "is_page_content": True,
+                "metadata": get_website_in_root_website_metadata(website, version),
+            },
+        )
+        backend = api.get_sync_backend(website=root_website)
+        backend.sync_all_content_to_backend(
+            query_set=WebsiteContent.objects.filter(text_id=website_content.text_id)
+        )
+        if not root_has_unpublished:
+            api.publish_website(root_website.name, version, trigger_pipeline=False)
```

Comment:
One potential complication if the root site pipeline should trigger, but the pipeline for a new individual site hasn't completed yet (or fails), is that the root site pipeline might fail because it can't find the json for that site?  I got this error after publishing a new local site then manually publishing the root site:

```
ERROR 2023/03/16 14:59:39 Failed to fetch new course info from https://ocw.mit.edu/courses/3-16-2-march-16-site-2-summer-2027/data.json
```

Reply:
> with `trigger_pipeline=False`, the git repo for the root site will be synced but it won't actually be published. Is that the desired behavior? The issue seems to infer that it should be published if there are no other changes.

The idea here is to get the markdown into Github, but the pipeline to build the live root website does not need to be triggered. Since the markdown files generated during this process have:

```
_build:
  list: true
  render: false
```

This means they will show up in lists during the Hugo build such as `site.Pages`, but no HTML or JSON will show up in the output based on them. This content will be used to build a paginated list of courses in an offline `ocw-www` site.



> One potential complication if the root site pipeline should trigger, but the pipeline for a new individual site hasn't completed yet (or fails), is that the root site pipeline might fail because it can't find the json for that site? I got this error after publishing a new local site then manually publishing the root site:
> 
> ```
> ERROR 2023/03/16 14:59:39 Failed to fetch new course info from https://ocw.mit.edu/courses/3-16-2-march-16-site-2-summer-2027/data.json
> ```

As far as this issue, can you try reproducing it on `master`? I don't believe it is related to this work. What I described above has nothing to do with the JSON that's fetched at build time by the live `ocw-www` site. When an individual course site is built, `data.json` files are generated and deployed as part of the build and the `www` theme queries these during its build based on an API response from `ocw-studio` about which courses have been published. What you're experiencing here is that you likely have `STATIC_API_URL` set to https://ocw.mit.edu/ in your env file. This gets pushed up to pipelines in your local Concourse and when building `ocw-www`, this is where these JSON files will be requested from. Since this course only exists in your local database, the request will fail every time. If you set `STATIC_API_URL` to http://10.1.0.102:8045/ then in theory you could have a pipeline hit your locally running `ocw-studio` nginx for this data, but unless you have published every site that it would be looking for then you will get errors like you posted above.